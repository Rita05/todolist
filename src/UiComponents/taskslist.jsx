const TasksList=(props)=>{
    return(
    <div className="flex flex-col justify-center items-center space-y-5 m-20px">
        {  
            props.tasks?.map(elem=>{
                return (
                <div key={elem.id} className="flex justify-between w-27.5rem px-4 py-2 bg-white">
                <div className="flex justify-start items-center space-x-5">
                    <input type='checkbox' className='pt-4'/>
                    <span className="text-center">{elem.taskName}</span>
                </div>
                    <button className='bg-teal-400 w-4rem h-2.5rem rounded-md' onClick={()=>props.onRemoveTask(elem.id)}>Delete</button>
                </div>
                )
            })
        }
    
    </div>)
}
export default TasksList
