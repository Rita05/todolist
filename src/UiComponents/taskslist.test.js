import { render } from "@testing-library/react"
import renderer from 'react-test-renderer'
import {TasksListDefault} from './taskslist.stories'
import TasksList from './taskslist'

describe("Testing render of taskslist component", () => {
  it("test render", () => {
    const props = {
      tasks: [
        {
          id: 1, 
          taskName: "Buy Products"
        },
        {
            id: 2, 
            taskName: "meet with friends"
        },
      ],
      onRemoveTask: () => {}, 
    }
    render(<TasksList {...props}/>);
    
  })
  it('render taskslist storybook component', ()=>{
    render(<TasksListDefault/>)
  })
  it('render correctly input task from', () => {
    const TasksListComponent = renderer.create(<TasksListDefault/>).toJSON();
    expect(TasksListComponent).toMatchSnapshot();
  });
})


