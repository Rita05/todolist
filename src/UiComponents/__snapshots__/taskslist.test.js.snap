// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render of taskslist component render correctly input task from 1`] = `
<div
  className="flex flex-col justify-center items-center space-y-5 m-20px"
/>
`;
